modules/serverless-blue-green-deploy.adoc

```diff
@@ -0,0 +1,112 @@
+
+.Procedure
+
+. Create and deploy an app as a Knative service.
+. Find the name of the first revision that was created when you deployed the service, by viewing the output from the following command:
++
+[source,terminal]
+----
+$ oc get configurations <service_name> -o=jsonpath='{.status.latestCreatedRevisionName}'
+----
+. Add the following YAML to the service `spec` to send inbound traffic to the revision:
++
+.Example Route
+[source,yaml]
+----
+...
+spec:
+  traffic:
+    - revisionName: <first_revision_name>
+      percent: 100 # All traffic goes to this revision
+...
+----
+. Verify that you can view your app at the URL output you get from running the following command:
++
+[source,terminal]
+----
+$ oc get route <route_name>
+----
+. Deploy a second revision of your app by modifying at least one field in the `template` spec of the service. For example, you can modify the `image` of the service, or an `env` environment variable.
+. Redeploy the service by applying the updated `Service` resource. You can do this by applying the service YAML file, or by using the `kn service update` command if you have installed the `kn` CLI.
+. Find the name of the second, latest revision that was created when you redeployed the service, by running the command:
++
+[source,terminal]
+----
+$ oc get configurations <service_name> -o=jsonpath='{.status.latestCreatedRevisionName}'
+----
++
+At this point, both the first and second revisions of the service are deployed and running.
+. Update your existing route to create a new, test endpoint for the second revision, while still sending all other traffic to the first revision:
++
+.Example of updated route
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Route
+metadata:
+  name: <route_name>
+  namespace: default
+spec:
+  traffic:
+    - revisionName: <first_revision_name>
+      percent: 100 # All traffic is still being routed to the first revision
+    - revisionName: <second_revision_name>
+      percent: 0 # 0% of traffic routed to the second revision
+      tag: v2 # A named route
+----
++
+After you redeploy this route by reapplying the YAML resource, the second revision of the app is now staged. No traffic is routed to the second revision at the main URL, and Knative creates a new route named `v2` for testing the newly deployed revision.
+. Get the URL of the new route for the second revision, by running the command:
++
+[source,terminal]
+----
+$ oc get route <route_name> --output jsonpath="{.status.traffic[*].url}"
+----
++
+You can use this URL to validate that the new version of the app is behaving as expected before you route any traffic to it.
+. Update your existing route resource again, so that 50% of traffic is being sent to the first revision, and 50% is being sent to the second revision:
++
+.Example of updated route
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Route
+metadata:
+  name: <route_name>
+  namespace: default
```

Comment:
```suggestion
. Update your existing service resource again, so that 50% of traffic is being sent to the first revision, and 50% is being sent to the second revision:
+
.Example of updated service
[source,yaml]
----
```